doc/matter/index.rst

```diff
@@ -16,6 +41,6 @@ To read about Matter in the nRF Connect SDK, see `Matter protocol <https://devel
    BUILDING
    nrfconnect_*
    android_*
-   openthread_*
+   openthread_rcp_nrf_dongle
```

Comment:
Let's remove this one as well. The instruction exists in NCS's Thread Border Router one.